lib/Model/Binary.cpp

```diff
@@ -100,6 +100,44 @@ bool Binary::verifyTypes(bool Assert) const {
   return verifyTypes(VH);
 }
 
+static bool beginsWithReservedPrefix(StringRef Name) {
+  return Name.startswith("unnamed_") or Name.startswith("revng_");
```

Comment:
The `"revng_"` prefix is needed because the `model-to-header` tool I'm working on will need to generate artificial type declarations for e.g. return types of `RawFunctionType`s.

For those I'm currently using `"revng_"` as a prefix, because `"unnamed_"` did not strike me as appropriate, given that I was generating them giving them a name. Also, I could go for `"unnamed_"` for them as well, but this will bring the unicity problem back. If a user defines a `model::Type` with a `CustomName` set to `revng_rawfunction_someid_return_type`, this will collide with the name we automatically generate in `model-to-header`. Using 2 namespaces (prefixes) ensures that the `"unnamed_"` guys never conflict with `"revng_"` guys.

Reply:
I see, however `revng_` is a bit generic. What about `artificial_`?

Reply:
My idea is that using `revng_` should minimize the chances that someone wants to use our reserved prefix, while at the same time clearly conveying the idea that a given thing is generated by `revng`. `artificial_` conveys the same idea, but it looks like a prefix that a user could want to use for something. `revng_` less so.

Random idea. Perhaps we should only reserve prefixes that look totally random and ugly, like e.g. `zyxwabcd_`. I expect that no sane person would want to use them for real, hence chances of conflicts are minimal.

But totally random prefixes look really ugly to see in generated code, and they don't convey any meaning. So they're probably a bad idea. At least `revng_` is not so ugly and I don't expect users to want to use that, but I might be wrong. Likewise, I deem `unnamed_` unlikely to be a prefix that people want to use for a name. If it has a name it's not unnamed after all.

So, if totally random prefixes look too ugly, we can go for `unnamed_` (namespace for unnamed types) and `artificial_` (namespace for stuff generated by `revng` that is NOT in the model, if you really don't like `revng_`).

Thoughts?

Reply:
Ok we converged on `unnamed_`, with subnamespaces where necessary.

Reply:
Done.